services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/LinkServiceImpl.java

```diff
@@ -181,4 +193,25 @@ public BatchResults delLinkProps(List<LinkPropsDto> linkPropsList) {
 
         return new BatchResults(errors.size(), successCount, errors);
     }
+
+    @Override
+    public List<FlowPayload> getFlowsForLink(SwitchId srcSwitch, Integer srcPort, SwitchId dstSwitch, Integer dstPort) {
+        final String correlationId = RequestCorrelationId.getId();
+        logger.debug("Get all flows for a particular link request processing");
+        GetFlowIdsForLinkRequest data = new GetFlowIdsForLinkRequest(new NetworkEndpointMask(srcSwitch, srcPort),
+                new NetworkEndpointMask(dstSwitch, dstPort), correlationId);
+        CommandMessage request = new CommandMessage(data, System.currentTimeMillis(), correlationId, Destination.WFM);
+        messageProducer.send(nbworkerTopic, request);
+
+        Message message = (Message) messageConsumer.poll(correlationId);
+        FlowsResponse flowsResponse = (FlowsResponse) validateInfoMessage(request, message, correlationId);
+
+        logger.debug("Received {} flow ids", flowsResponse.getFlowIds().size());
+
+        List<FlowPayload> flows = flowService.getFlows();
```

Comment:
Fetching hundreds(in near future thousands) flows and filter out most of them is bad idea. Even more - I believe we will need pagination for "getAllFlows" call in near future. And all such "use cases" will collide with this change.

Read only flows listed in `flowResponse`.